HuliWebShop/src/main/java/huli/example/huliwebshop/services/OrderServiceImpl.java

```diff
@@ -32,21 +33,21 @@ public Long createOrder(OrderDTO orderDTO) {
             throw new IllegalArgumentException("User not found with ID: " + orderDTO.getUserId());
         }
 
-        Set<Product> products = new HashSet<>();
-        for (Long productId : orderDTO.getProductIds()) {
+        Set<OrderProduct> products = new HashSet<>();
+        /*for (Long productId : orderDTO.getProductIds()) {
             Product product = iProductRepository.findById(productId).orElse(null);
             if (product != null) {
                 products.add(product);
             }
-        }
+        }*/
 
         if (products.isEmpty()) {
             throw new IllegalArgumentException("No valid products found in the order");
         }
 
         Order order = new Order();
         order.setUser(user);
-        order.setProducts(products);
+        order.setOrderProducts(products);
         order.setTotalPrice(orderDTO.getTotalPrice());
```

Comment:
getting the total price is a security risk as well, this happend with BKK exactly :)
if the user can post a price whatever they choose, they will post 1 HUF and we will simply acknowledge it :) 
only the productids and quantities should be presented in the dto (since that's the user's choice, what they want to buy how how many of each), everything else should come from the db.